packages/flutter_tools/lib/src/base/process.dart

```diff
@@ -607,3 +609,387 @@ class _DefaultProcessUtils implements ProcessUtils {
     }
   }
 }
+
```

Comment:
I've only made a small adjustment to the API of ProcessManager, changing it to take a List<String> instead of a List<dynamic>.  I did slightly adjust the ArgumentError thrown when we fail to resolve an executable.

My plan is to follow up with PRs to adjust that behavior which would be significantly smaller in scope